chemprop/v2/cli/train.py

```diff
@@ -801,7 +802,7 @@ def main(args):
 
     readout_ffn = Factory.build(
         readout_cls,
-        input_dim=mp_block.output_dim,
```

Comment:
I guess this is a separate issue.

I ran a test and found that the `input_dim` is still `300` despite the change here. I found that the factory method does not work as expected for `RegressionFFN` (while it works for `BinaryClassificationFFN`). I think the issue is due to the fact that `RegressionFFN` has `__init__` redefined, and different from `ReadoutFFNBase`. Then, in `build`:

https://github.com/chemprop/chemprop/blob/e777044bfc46bfb14927a53bc89b366dfa17acab/chemprop/v2/utils/registry.py#L39-L46

the obtained `sig.parameters.keys()` is actually `odict_keys(['args', 'loc', 'scale', 'kwargs'])` that doesn't include `input_dim` and other arguments...

Reply:
good catch! I've changed all `ReadoutFFN` (sub)classes to now use explicit `__init__` signatures to fix this.